src/linked_list.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""Create a singly linked list."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """Create a node to store data."""
+
+    def __init__(self, value, pointer=None):
+        """Create an instance of a Node."""
+        self.value = value
+        self.pointer = pointer
+
+
+class LinkedList(object):
+    """Create a link list class to store strings of data."""
+
+    def __init__(self, value_list=None):
+        """Create an instance of a singly-linked list."""
+        self.length = 0
+        self.header = None
+        if hasattr(value_list, '__iter__'):
+            for value in value_list:
+                self.push(value)
+        elif value_list:
+            self.push(value_list)
+
+    def __len__(self):
+        """Return the length of the linked list for the built in len."""
+        return self.length
+
+    def push(self, value):
+        """Add a new node to the head of the linked list."""
+        temp_node = self.header
+        new_node = Node(value, temp_node)
+        self.header = new_node
+        self.length += 1
+
+    def size(self):
+        """Return the length of the linked list."""
+        return len(self)
+
+    def pop(self):
+        """Remove the first value off the head of the list and return it."""
+        if self.length:
+            pop_node = self.header
+            self.header = pop_node.pointer
+            self.length -= 1
+            return pop_node.value
```

Comment:
good.  I would probably have allowed an error to happen on popping from an empty list.  It would be more informative than what happens here.  Do you know what the result is when you pop an empty list here?